diff --git a/compiler/typecheck/TcSplice.lhs b/compiler/typecheck/TcSplice.lhs
index 5a55d25..d4a5cb3 100644
--- a/compiler/typecheck/TcSplice.lhs
+++ b/compiler/typecheck/TcSplice.lhs
@@ -1663,11 +1663,12 @@ lookupThAnnLookup (TH.AnnLookupModule (TH.Module pn mn))
 reifyAnnotations :: Data a => TH.AnnLookup -> TcM [a]
 reifyAnnotations th_nm
   = do { name <- lookupThAnnLookup th_nm
-       ; eps <- getEps
+       ; hscEnv <- getTopEnv
+       ; epsHptAnnEnv <- liftIO $ prepareAnnotations hscEnv Nothing
        ; tcg <- getGblEnv
-       ; let epsAnns = findAnns deserializeWithData (eps_ann_env eps) name
-       ; let envAnns = findAnns deserializeWithData (tcg_ann_env tcg) name
-       ; return (envAnns ++ epsAnns) }
+       ; let epsHptAnns = findAnns deserializeWithData epsHptAnnEnv name
+       ; let tcgAnns = findAnns deserializeWithData (tcg_ann_env tcg) name
+       ; return (epsHptAnns ++ tcgAnns) }
 
 ------------------------------
 modToTHMod :: Module -> TH.Module
